seed/seed20.go

```diff
@@ -353,6 +425,91 @@ func (e *noSnapDeclarationError) Error() string {
 	return fmt.Sprintf("cannot find snap-declaration for snap name: %s", e.snapRef.SnapName())
 }
 
+type errorComponentNotInSeed struct {
+	error
+}
+
+func (s *seed20) lookupVerifiedComponent(cref naming.ComponentRef, snapRev snap.Revision, snapID, snapProvenance string, snapsDir string, handler ContainerHandler, tm timings.Measurer) (Component, error) {
+	snapName := cref.SnapName
+	compName := cref.ComponentName
+
+	resKey := resourceKey{snapID: snapID, name: compName}
+	resRev, ok := s.resRevByResKey[resKey]
+	if !ok {
+		// No assertions might be ok if the component is optional, the
+		// caller should check for this error type in that case.
+		return Component{}, errorComponentNotInSeed{
+			fmt.Errorf("resource revision assertion not found for %s", compName)}
+	}
+	resPair, ok := s.resPairByResKey[resKey]
+	if !ok {
+		return Component{}, errorComponentNotInSeed{
+			fmt.Errorf("resource pair assertion not found for %s", compName)}
```

Comment:
but if we found the resRev this more of an internal error, no? given the checks when we create the maps?